pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -392,30 +398,61 @@ void processArguments() throws Exception {
 
         protected void validateFunctionConfigs(FunctionConfig functionConfig) {
 
-            if (functionConfig.getJar() != null && functionConfig.getPy() != null) {
+            if (isNotBlank(functionConfig.getJar()) && isNotBlank(functionConfig.getPy())) {
                 throw new ParameterException("Either a Java jar or a Python file needs to"
                         + " be specified for the function. Cannot specify both.");
             }
 
-            if (functionConfig.getJar() == null && functionConfig.getPy() == null) {
+            if (isBlank(functionConfig.getJar()) && isBlank(functionConfig.getPy())) {
                 throw new ParameterException("Either a Java jar or a Python file needs to"
                         + " be specified for the function. Please specify one.");
             }
 
-            if (!fileExists(userCodeFile)) {
-                throw new ParameterException("File " + userCodeFile + " does not exist");
+            boolean isJarPathUrl = isNotBlank(functionConfig.getJar()) && Utils.isFunctionPackageUrlSupported(functionConfig.getJar());
```

Comment:
why are we doing this just for jar?(and not userCodeFile?)

Reply:
Right now, with this PR, we are adding url support for java-type and server-side validation happens for java-type only. `userCodeFile` can be python as well so, we are explicitly doing it for jar. we can use `userCodeFile` with PR where we can support python as well.